apis/v1beta1/types.go

```diff
@@ -34,12 +34,34 @@ type ProviderConfigSpec struct {
 // ProviderCredentials required to authenticate.
 type ProviderCredentials struct {
 	// Source of the provider credentials.
-	// +kubebuilder:validation:Enum=None;Secret;AccessToken;InjectedIdentity;Environment;Filesystem
+	// +kubebuilder:validation:Enum=None;Secret;AccessToken;InjectedIdentity;Environment;Filesystem;Upbound
 	Source xpv1.CredentialsSource `json:"source"`
 
+	// Upbound defines the options for authenticating using Upbound as an
+	// identity provider.
+	Upbound *Upbound `json:"upbound,omitempty"`
+
 	xpv1.CommonCredentialSelectors `json:",inline"`
 }
 
+// Upbound defines the options for authenticating using Upbound as an identity
+// provider.
+type Upbound struct {
+	// Federation is the configuration for federated identity.
+	Federation *Federation `json:"federation"`
+}
+
+// Federation defines the configuration for federated identity from an external
+// provider.
+type Federation struct {
+	// ProviderID is the fully-qualified identifier for the identity provider on
+	// GCP. The format is
+	// `project/<project-id>/locations/global/workloadIdentityPools/<identity-pool>/providers/<identity-provider>`.
+	ProviderID string `json:"providerID"`
```

Comment:
nit: Should we add some validation markers for these (at least to make sure they are non-empty)?

Reply:
@ulucinar I would love to -- what is y'all's current standard here? Just using validation such as min-length or have y'all started incorporating CEL experessions?

Reply:
I believe just a length check would be great for now if you also think it's worth. I don't know of any examples of regex checking or CEL-expressions yet. Unfortunately, we do not have a high standard here yet :)

Reply:
@ulucinar sounds reasonable -- thanks for the suggestion and context!

Reply:
updated!